src/System.Linq.Parallel/tests/QueryOperators/AggregateTests.cs

```diff
@@ -0,0 +1,199 @@
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Result_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate_Result(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Accumulator(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            int actual = query.Aggregate(
+                0,
+                (accumulator, x) => accumulator + x,
+                (left, right) => left + right,
+                result => result);
+            Assert.Equal(Functions.SumRange(0, count), actual);
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Accumulator_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate_Accumulator(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_SeedFunction(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            int actual = query.Aggregate(
+                () => 0,
+                (accumulator, x) => accumulator + x,
+                (left, right) => left + right,
+                result => result);
+            Assert.Equal(Functions.SumRange(0, count), actual);
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_SeedFunction_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate_SeedFunction(labelled, count);
+        }
+
+        [Fact]
+        public static void Aggregate_InvalidOperationException()
+        {
+            Assert.Throws<InvalidOperationException>(() => ParallelEnumerable.Empty<int>().Aggregate((i, j) => i));
+            // All other invocations return the seed value.
+            Assert.Equal(-1, ParallelEnumerable.Empty<int>().Aggregate(-1, (i, j) => i + j));
+            Assert.Equal(-1, ParallelEnumerable.Empty<int>().Aggregate(-1, (i, j) => i + j, i => i));
+            Assert.Equal(-1, ParallelEnumerable.Empty<int>().Aggregate(-1, (i, j) => i + j, (i, j) => i + j, i => i));
+            Assert.Equal(-1, ParallelEnumerable.Empty<int>().Aggregate(() => -1, (i, j) => i + j, (i, j) => i + j, i => i));
+        }
+
+        [Fact]
+        public static void Aggregate_OperationCancelledException()
```

Comment:
The combinatorial tests don't check for `InvalidOperationException` (mostly because it doesn't return an empty source) or `OperationCanceledException` (yet - stuff I'm writing after I get this first batch done, and even then it's throwing one level down).